pkg/openstack/common.go

```diff
@@ -647,6 +666,23 @@ func (ed *EndpointDetail) CreateRoute(
 			return ctrlResult, nil
 		}
 
+		// Delete the issued certificate if it exists and custom cert secret or direct TLS data was provided
+		if ed.Route.TLS.SecretName != nil || hasCertInOverrideSpec(ed.Route.OverrideSpec) {
+			cert := certmanager.NewCertificate(
+				&certmgrv1.Certificate{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      ed.Route.TLS.CertName,
+						Namespace: ed.Namespace,
+					},
+				},
+				5*time.Second,
+			)
+			err := cert.Delete(ctx, helper)
+			if err != nil {
+				helper.GetLogger().Error(err, fmt.Sprintf("Failed to delete unused route certificate %s", ed.Route.TLS.CertName))
```

Comment:
its a duplicate error message as we already get the one from https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/certmanager/certificate.go#L160, do we need this?